bazel-mode.el

```diff
@@ -21,6 +21,31 @@
 ;;
 ;;; Code:
 
+(defcustom buildifier-cmd "buildifier"
+  "Filename of buildifier executable."
+  :type 'string
+  :group 'bazel-mode)
+
+(defun buildifier ()
+  "Format current buffer using buildifier."
+  (interactive)
+  (let ((build-file-contents (buffer-string))
+		(input-buffer (current-buffer))
+		(orig-point (point)))
+	(with-current-buffer (get-buffer-create "*buildifier*")
+	  (erase-buffer)
+	  (insert build-file-contents)
+	  (let ((return-code
+			 (call-process-region (point-min) (point-max) buildifier-cmd t t)))
+		(unwind-protect
+		  (if (zerop return-code)
+			  (progn
+				(copy-to-buffer input-buffer (point-min) (point-max))
```

Comment:
Please use `replace-buffer-contents` so that markers etc. stay intact.

Reply:
Done.